### PR TITLE
Add rekey to to inner tx

### PR DESCRIPTION
**Pull request by vuvth**
## Proposed Changes

+ add Inner Transaction rekeyTo
+ uint and integration tests.

### CODE REVIEW

File: packages/runtime/src/lib/itxn.ts [hunk 1]
```diff
@@ -72,6 +72,10 @@ export function setInnerTxField (
     const accountState = interpreter.getAccount(assertedVal, line);
     txValue = Buffer.from(decodeAddress(accountState.address).publicKey);
   }
+  // begin supported RekeyTo form teal version 6
+  if (field === 'RekeyTo' && interpreter.tealVersion >= 6) {
+    txValue = op.assertAlgorandAddress(val, line);
+  }
```
Comment (ratik21) [inline packages/runtime/src/lib/itxn.ts:78]: Can we put the fields into an array/set or something? What if we need to add more fields for more versions?

File: packages/runtime/test/fixtures/inner-transaction/assets/approval-rekey.teal [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+itxn_field RekeyTo
+int 1000
+itxn_field Fee
+itxn_submit
+int 1
+return
+
+on_creation:
+int 1
+return
```
Comment (ratik21) [inline packages/runtime/test/fixtures/inner-transaction/assets/approval-rekey.teal:24]: note: please try to add in PyTEAL, it's easier to read

File: packages/runtime/test/integration/inner-transaction/rekey.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  it("should rekey to another account address", function () {
+    // rekey to master account
+    const paymentTxParams = {
+      ...appCallParams,
+      accounts: [master.address]
+    };
+    runtime.executeTx(paymentTxParams);
+    syncAccounts();
+    assert.equal(appAccount.getSpendAddress(), master.address);
```
Comment (ratik21) [inline packages/runtime/test/integration/inner-transaction/rekey.ts:75]: We should also add an algotransfer transaction now, to test the "rekeying" of app account. i.e inner transaction "from" smart contract fails, but algo transfer tx "from master" should pass.

File: packages/runtime/test/src/interpreter/inner-transaction.ts [hunk 4]
```diff
@@ -569,9 +561,12 @@ describe("TEALv5: Inner Transactions", function () {
...
     it(`should pass: after increasing app account's balance`, function () {
       const acc = interpreter.runtime.ctx.state.accounts.get(appAccAddr);
-      if (acc) { acc.amount = 1000000n; }
+      if (acc) {
+        acc.amount = 1000000n;
+      }
 
-      const teal = `
+      const teal =
+        `
```
Comment (ratik21) [inline packages/runtime/test/src/interpreter/inner-transaction.ts:569]: is your editor doing this?

Comment (vuvth) [reply packages/runtime/test/src/interpreter/inner-transaction.ts:569]: > is your editor doing this?

Yes, It's automatic, maybe I should disable it. I remember husky auto format for us but now it not. How about it on your site?

File: packages/runtime/test/fixtures/inner-transaction/assets/approval-rekey.teal (hunk 2 above)
Comment (vuvth) [reply packages/runtime/test/fixtures/inner-transaction/assets/approval-rekey.teal:24]: > note: please try to add in PyTEAL, it's easier to read
We can't do it now. PyTEAL is not support tealv6 on master branch now.

Comment (vuvth) [reply packages/runtime/test/fixtures/inner-transaction/assets/approval-rekey.teal:24]: > note: please try to add in PyTEAL, it's easier to read

 We can't do it now. PyTEAL is not support tealv6 on master branch.

File: packages/runtime/test/integration/inner-transaction/rekey.ts (hunk 3 above)
Comment (vuvth) [reply packages/runtime/test/integration/inner-transaction/rekey.ts:75]: > We should also add an algotransfer transaction now, to test the "rekeying" of app account. i.e inner transaction "from" smart contract fails, but algo transfer tx "from master" should pass.

I think it has relate with this task. So i will do it in another PR.

**Review by ratik21 — APPROVED** [review]

**Pull request merged**